crates/api/src/op_store.rs

```diff
@@ -77,12 +77,19 @@ pub trait OpStore: 'static + Send + Sync + std::fmt::Debug {
     ///
     /// This must be the timestamp of the op, not the time that we saw the op or chose to store it.
     /// The returned ops must be ordered by timestamp, ascending.
+    ///
+    /// # Returns
+    ///
+    /// - As many op ids as can be returned within the `limit_bytes` limit, within the arc and time
+    ///   bounds.
+    /// - The total size of the op data that is pointed to by the returned op ids.
     fn retrieve_op_hashes_in_time_slice(
         &self,
         arc: DhtArc,
         start: Timestamp,
         end: Timestamp,
-    ) -> BoxFuture<'_, K2Result<Vec<OpId>>>;
+        limit_bytes: Option<u32>,
```

Comment:
Optional limit because this method is used in places where we don't want a limit.